### ISSUE TITLE
Data classes for event wrappers

### ISSUE THREAD

**Issue by octa-one** [ReactiveCircus/FlowBinding#112]
What do you think about converting "event wrapper" classes to data classes?
(TextChangeEvent, ScrollEvent, EditorActionEvent, etc)

I see at least two improvements:
* `distinctUntilChanged` will work correctly with overloaded `equals`
* ability to use destructuring declaration in `collect` lambdas, for example:
```kotlin
// w/o data class
textView.textChangeEvents().collect {
    doSmth(it.view, it.text)
}

// with data class
textView.textChangeEvents().collect { (view, text, _, _, _) ->
    doSmth(view, text)
}
```

**Comment by ychescale9** [comment]
I agree it's a good idea and thanks for the destructuring example.

This change will change the ABI although I'm not too concerned since converting to a data class should be binary compatible change.

**Comment by octa-one** [comment]
Great, thanks for the quick response!

**Comment by ychescale9** [comment]
This change is now available in `1.0.0-beta02`.
